python/lsst/meas/extensions/gaap/_gaap.py

```diff
@@ -551,8 +551,18 @@ def _gaussianizeAndMeasure(self, measRecord: lsst.afw.table.SourceRecord,
             raise measBase.FatalAlgorithmError("No PSF in exposure")
         wcs = exposure.getWcs()
 
-        psfSigma = psf.computeShape(center).getDeterminantRadius()
-        errorCollection = dict()
+        psfShape = psf.computeShape(center)
+        if not (psfShape.getDeterminantRadius() > 0):
+            self.log.debug("PSF shape has non-positive determinant radius at (%f, %f)", center.x, center.y)
+
+        psfSigma = psfShape.getTraceRadius()
+        if not (psfSigma > 0):  # This captures nan
+            errorCollection = {str(scalingFactor): measBase.MeasurementError("PSF size could not be measured")
+                               for scalingFactor in self.config.scalingFactor}
+            raise GaapConvolutionError(errorCollection)
+        else:
+            errorCollection = {}
```

Comment:
This may be personal preference, but I think the syntax you used before (`dict()`) reads better than the suggested change (`{}`). Completely up to you however!